frameworks/move-stdlib/sources/macros.move

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This module holds shared implementation of macros used in `std`
+module std::macros {
+    public macro fun num_max($x: _, $y: _): _ {
```

Comment:
Do macros support arguments of arbitrary types?

Reply:
See the above doc.

> Type inference holes _ on type annotations
> 
> With type directed programming, often you need to annotate a variable declaration or provide type arguments. But, sometimes you really only need to annotate on specific type, but the other types can be inferred. _ will be added to allow that type to still be inferred, even when other parts of the type are annotated. For example
> 
> dynamic_field::borrow_mut<address, Coin<SUI>>(&mut id, owner)
> 
> could be rewritten as
> 
> dynamic_field::borrow_mut<_, Coin<SUI>>(&mut id, owner)
> 
> where the _ would be inferred as address